front/components/data_source/RequestDataSourcesModal.tsx

```diff
@@ -76,37 +73,41 @@ export function RequestDataSourcesModal({
                 )}
               </DropdownMenu.Button>
               <DropdownMenu.Items width={180}>
-                {filteredDataSourceIntegrations.map((ds) => (
-                  <DropdownMenu.Item
-                    key={ds.dataSourceName}
-                    label={ds.name}
-                    onClick={() => setSelectedDataSourceIntegration(ds)}
-                    icon={
-                      CONNECTOR_CONFIGURATIONS[ds.connectorProvider]
-                        .logoComponent
-                    }
-                  />
-                ))}
+                {dataSources.map(
+                  (dataSource) =>
+                    dataSource.connectorProvider && (
+                      <DropdownMenu.Item
+                        key={dataSource.name}
+                        label={
+                          CONNECTOR_CONFIGURATIONS[dataSource.connectorProvider]
+                            .name
+                        }
+                        onClick={() => setSelectedDataSource(dataSource)}
+                        icon={
+                          CONNECTOR_CONFIGURATIONS[dataSource.connectorProvider]
+                            .logoComponent
+                        }
+                      />
+                    )
+                )}
               </DropdownMenu.Items>
             </DropdownMenu>
           )}
         </div>
 
-        {selectedDataSourceIntegration && (
+        {selectedDataSource && (
           <div>
             <p className="s-mb-2 s-text-sm s-text-element-700">
               The administrator for{" "}
-              {
-                CONNECTOR_CONFIGURATIONS[
-                  selectedDataSourceIntegration.connectorProvider
-                ].name
-              }{" "}
-              is {selectedDataSourceIntegration.editedByUser?.fullName}. Send an
-              email to {selectedDataSourceIntegration.editedByUser?.fullName},
-              explaining your request.
+              {selectedDataSource.connectorProvider &&
+                CONNECTOR_CONFIGURATIONS[selectedDataSource.connectorProvider]
+                  .name}{" "}
+              is {selectedDataSource.editedByUser?.fullName}. Send an email to{" "}
+              {selectedDataSource.editedByUser?.fullName}, explaining your
+              request.
             </p>
             <TextArea
-              placeholder={`Hello ${selectedDataSourceIntegration.editedByUser?.fullName},`}
+              placeholder={`Hello ${selectedDataSource.editedByUser?.fullName},`}
```

Comment:
Note : editedByUser is not loaded yet, to be continued on https://github.com/dust-tt/dust/pull/6964